### ISSUE TITLE
Add an example of adding an external package to the demo project

### ISSUE THREAD

**Issue by freakboy3742** [beeware/beeware#75]
A recurring question on the bug tracker and chat room is confusion over why an app works in the `dev` environment, but doesn't work under `briefcase run` (especially on mobile).

The fix is almost always easy - add your requirements to `pyproject.toml`, in the `requires` declaration. 

This isn't ever explicitly called out in the tutorial - but it should be. `requires` is the one field that almost every project will need to interact with, so we should point this out during the tutorial.

**Describe the solution you'd like**

An addition to the tutorial that requires the addition of an external module installed from pip. 

The "Updating dependencies and icons" step on tutorial 4 potentially provides a good entry to this discussion; having modified the app to include a dialog, we could introduce another code change that depends on some external code module. requests/httpx would be a good option - adding some value retrieved from a REST API (say, the most toga version on PyPI?) to the content output by the dialog.

This has the added benefit that it shows off a feature that is likely to be a common requirement for real-world apps.

**Comment by supagu** [comment]
I am having exactly this problem.
I am setting up the virtual env like so:

```
python3 -m venv beeware-venv

# activate the virtual environment
source beeware-venv/bin/activate

# install deps into the virtual env
python -m pip install briefcase
python -m pip install broadlink
```

then in my pyproject.toml I have:

```
requires = [
    'broadlink>=0.17.0'
]
```

I then update android
then try to run android I get the following errors:

`ModuleNotFoundError: No module named 'broadlink'`

Followed by:

```
2021-04-01 13:06:21.709 9064-9064/io.bitshift.schmarts.schmarts E/AndroidRuntime: FATAL EXCEPTION: main
    Process: io.bitshift.schmarts.schmarts, PID: 9064
    java.lang.RuntimeException: Unable to start activity ComponentInfo{io.bitshift.schmarts.schmarts/org.beeware.android.MainActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void org.beeware.android.IPythonApp.onCreate()' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3449)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void org.beeware.android.IPythonApp.onCreate()' on a null object reference
```

Am I missing a step?

**Comment by freakboy3742** [comment]
If you added `broadlink` *after* invoking `briefcase create`, then you'll need to run `briefcase update -d` to tell Briefcase to pick up the new dependency. 

If you added `broadlink` requires entry *before* generating your project with `briefcase create` (or you've run `briefcase update -d` then you've done the right thing - the problem will almost certainly be that broadlink (or one of it's dependencies) has a binary module. This is beeware/briefcase#471 - it's a known issue we'd like to address, but it's not simple to fix.

**Comment by freakboy3742** [comment]
Fixed by #126.
